core/build.gradle.kts

```diff
@@ -30,6 +30,8 @@ dependencies {
   annotationProcessor(libs.lombok)
   compileOnly(libs.lombok)
 
+  compileOnly(libs.servlet) // fix error-prone compile error
```

Comment:
I can't figure out why the core needs the servlet dependencies, but this can fix the compile error.
```
error: cannot access HttpServlet
  class file for javax.servlet.http.HttpServlet not found
  Consult the following stack trace for details.
  com.sun.tools.javac.code.Symbol$CompletionFailure: class file for javax.servlet.http.HttpServlet not found
```

Reply:
It is weird to include this, can you please figure out why error-prone need this.

Reply:
@jerryshao I think it is because `MetricsServlet` from MetricsSystem.
https://github.com/datastrato/gravitino/blob/main/core/src/main/java/com/datastrato/gravitino/metrics/MetricsSystem.java#L176

Reply:
@FANNG1 can you please check this?

Reply:
ok

Reply:
the core package doesn't use `javax.servlet.http.HttpServlet` directly, it's confusing for the compiling failure.

Reply:
I don't have a better solution, so I think it is acceptable for now.

Reply:
I think it is related to the mechanism of `error-prone`. The `error-prone` will parse the code to `AST`. When it matches a new class, it will try to find it in the classpath and create a ClassTree. When invoking `new MetricsServlet()`, it will parser the MetricsServer.java to do some check such as overloaded. But the superclass `HttpSevlet` doesn't exist in the current compile classpath. So it causes `class file not found`.

Reply:
If we remove the code about the `MetricsServlet`, then the issue will be gone. But we have to create the MetricsServlet in `server-common` which is not a good change. So I suggest we can keep this for now, because the scope of the servlet is `CompileOnly`, it will not impact. 
@jerryshao What do you think?

Reply:
> If we remove the code about the `MetricsServlet`, then the issue will be gone. But we have to create the MetricsServlet in `server-common` which is not a good change. So I suggest we can keep this for now, because the scope of the servlet is `CompileOnly`, it will not impact. @jerryshao What do you think?

I agree.